### PR TITLE
Put the right-click message context menu to the right

### PR DESCRIPTION
**Pull request by SimonBrandner**
Type: enhancement
Fixes https://github.com/vector-im/element-web/issues/21813
Notes: none

---
This change has no change notes, so will not be included in the changelog.

Preview: https://pr8339--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by SimonBrandner** [comment]
Tests failing for unrelated reasons

File: src/components/views/rooms/EventTile.tsx [hunk 1]
```diff
@@ -230,7 +230,7 @@ interface IState {
 
     // Position of the context menu
     contextMenu?: {
-        position: Pick<DOMRect, "right" | "top" | "bottom">;
+        position: Pick<DOMRect, | "top" | "left" | "bottom">;
```
Comment (robintown) [inline src/components/views/rooms/EventTile.tsx:233]: Extra `|` before `"top"`? or does that have a meaning I'm not aware of

File: src/components/views/rooms/EventTile.tsx [hunk 2]
```diff
@@ -1017,7 +1017,7 @@ export class UnwrappedEventTile extends React.Component<IProps, IState> {
 
         return (
             <MessageContextMenu
-                {...aboveLeftOf(this.state.contextMenu.position)}
+                {...aboveRightOf(this.state.contextMenu.position)}
```
Comment (robintown) [inline src/components/views/rooms/EventTile.tsx:1020]: Could this use an `alwaysBelowRightOf` placement instead? I don't think it's conventional to place context menus above the cursor.

**Review by robintown — COMMENTED** [review]
Thanks :)

File: src/components/views/rooms/EventTile.tsx (hunk 2 above)
Comment (SimonBrandner) [reply src/components/views/rooms/EventTile.tsx:1020]: This actually puts the menu above/below depending on which half of the screen you click. From what I gather the context-menu positioning system is somewhat flawed and might not allow what we actually want

Comment (SimonBrandner) [reply src/components/views/rooms/EventTile.tsx:1020]: https://user-images.githubusercontent.com/25768714/163706819-228388dc-1f62-45ed-8d23-62710e71d18b.mp4

Does this satisfy your needs or should I try to figure out something better?

Comment (robintown) [reply src/components/views/rooms/EventTile.tsx:1020]: Hm, right, I suppose that's fine

**Review by robintown — APPROVED** [review]

**Pull request merged**